# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaDecoderStateMachine.cpp
# Commit: f436cf1a01cc
# Full Hash: f436cf1a01cc3a36b703aa9c584a98d3246492f0
# Author: alwu <alwu@mozilla.com>
# Date: 2022-11-05 21:53:50
# Regressor Bug: 1262276
# File Overlap Count: 1
# Description:
#   Bug 1262276 - part17 : disable late time check for seamless looping. r=padenot
#   
#   This doesn't work for seamless looping because the video end time can
#   be larger than the media time. We would need to revisit these mechaniam
#   later to see if we can have a better way to skip to next frame.
# ==============================================================================

diff -r ee73611a2993 -r f436cf1a01cc dom/media/MediaDecoderStateMachine.cpp
--- a/dom/media/MediaDecoderStateMachine.cpp	Sat Nov 05 02:23:35 2022 +0000
+++ b/dom/media/MediaDecoderStateMachine.cpp	Sat Nov 05 02:23:35 2022 +0000
@@ -597,8 +597,11 @@
   }
 
   void HandleVideoDecoded(VideoData* aVideo) override {
+    // We only do this check when we're not looping, which can be known by
+    // checking the queue's offset.
     const auto currentTime = mMaster->GetMediaTime();
-    if (aVideo->GetEndTime() < currentTime) {
+    if (aVideo->GetEndTime() < currentTime &&
+        VideoQueue().GetOffset() == media::TimeUnit::Zero()) {
       if (!mVideoFirstLateTime) {
         mVideoFirstLateTime = Some(TimeStamp::Now());
       }
